compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingAttrs.td

```diff
@@ -142,6 +142,31 @@ def EncodingAttr :
   let genVerifyDecl = 1;
 }
 
+//===---------------------------------------------------------------------===//
+// encoding.matmul_k
+//===---------------------------------------------------------------------===//
+
+def MatmulKAttr : IREEEncoding_Attr<"MatmulK"> {
+  let mnemonic = "matmul_k";
+  let summary = [{An attribute that tracks reduction dimensions for matmul}];
+  let description = [{
+    The attribute is specialized for tracking the matmul reduction dimensions
+    only. The encoded dimensions are the reduction dimensions that will be
+    further used in a matmul.
+
+    Any encoding propagation that transforms reduction dimensions could result
+    in undefined behavior. Because it does not encode the transformations. The
+    information is missing in this context.
+  }];
+
+  let assemblyFormat = "`<` struct(params) `>`";
+
+  let parameters = (ins
+    "DenseI32ArrayAttr":$k_dims,
+    OptionalParameter<"ArrayAttr", "the serialized layouts, if present">:$layouts
```

Comment:
I'm thinking to introduce `iree_encoding.layout<>` attribute, and all the existing lowering use the encoding instead. Today, we encode everything in EncodingAttr, which is a generic encoding to me. Once the layout is resolved, all the other information is not needed. Thus, having a new `iree_encoding.layout` attribute makes sense to me. And we can reuse it for any other new encodings that can be on tensor types.

(I'll start the work after we land https://github.com/iree-org/iree/pull/20459. Because conflicts are pain.)

Reply:
I should just drop this parameter in the PR. I will introduce the new layout encoding attribute and complete the migration for encoding_attr later.

Reply:
+1 on the `iree_encoding.layout<>` attribute. It makes a lot of sense indeed to have a dedicated attribute for the resolved layout.

Reply:
Issue created: https://github.com/iree-org/iree/issues/20489